traitsui/testing/tester/wx/helpers.py

```diff
@@ -55,6 +55,83 @@ def mouse_click_checkbox(control, delay):
     control.ProcessWindowEvent(click_event)
 
 
+def mouse_click_combobox_or_choice(control, index, delay):
+    """ Performs a mouce click on either a wx combo box or a wx choice on the
+    entry at the given index.
+
+    Parameters
+    ----------
+    control : wxComboBox or wxChoice
+        The wx Object to be clicked.
+    index : int
+        The index of the item in the combobox/choice to be clicked
+    delay: int
+        Time delay (in ms) in which click will be performed.
+    """
+    if not control.IsEnabled():
+        return
+    wx.MilliSleep(delay)
+    if isinstance(control, wx.ComboBox):
+        click_event = wx.CommandEvent(
+            wx.wxEVT_COMMAND_COMBOBOX_SELECTED, control.GetId()
+        )
+    elif isinstance(control, wx.Choice):
+        click_event = wx.CommandEvent(
+            wx.wxEVT_COMMAND_CHOICE_SELECTED, control.GetId()
+        )
+    else:
+        raise TypeError
+    click_event.SetEventObject(control)
+    click_event.SetString(control.GetString(index))
+    control.SetSelection(index)
+    control.ProcessWindowEvent(click_event)
+
+
+def mouse_click_listbox(control, index, delay):
+    """Performs a mouce click on a wx list box on the entry at
+    the given index.
+
+    Parameters
+    ----------
+    control : wxListBox
+        The wx Object to be clicked.
+    index : int
+        The index of the item in the list box to be clicked
+    delay: int
+        Time delay (in ms) in which click will be performed.
+    """
+    if not control.IsEnabled():
+        return
+    wx.MilliSleep(delay)
+    click_event = wx.CommandEvent(
+        wx.wxEVT_COMMAND_LISTBOX_SELECTED, control.GetId()
+    )
+    click_event.SetEventObject(control)
+    control.SetSelection(index)
+    control.ProcessWindowEvent(click_event)
+
+
+def mouse_click_radiobutton(control, delay):
+    """ Performs a mouce click on a wx radio button.
+
+    Parameters
+    ----------
+    control : wxRadioButton
+        The wx Object to be clicked.
+    delay: int
+        Time delay (in ms) in which click will be performed.
+    """
+    if not control.IsEnabled():
+        return
+    wx.MilliSleep(delay)
+    click_event = wx.CommandEvent(
+        wx.wxEVT_COMMAND_RADIOBUTTON_SELECTED, control.GetId()
+    )
+    click_event.SetEventObject(control)
+    control.SetValue(not control.GetValue())
+    control.ProcessWindowEvent(click_event)
+
```

Comment:
I do like this idea for refactor.  However, a refactor like this could impact effectively every keyclick function in helpers.  It may be better to make this change as its own PR entirely, and just leave the repetition for now.

Reply:
I have opened a separate issue for this (#1194)

Reply:
I think we can limit the scope of this refactoring to only mouse_click_* function that uses this same pattern. In other words, it will affect the existing `mouse_click_checkbox` and no more. 

It is a judgment call whether one should split up refactoring into a separate PR, depending on the circumstances and the scope. A lot of times refactoring will be motivated by adding new code that repeats patterns in existing code, and the refactoring will change existing code. It would be acceptable to include related refactoring in a PR that adds features... unless when the scope got too big to be efficient. In this case, I'd be in favour of doing the refactoring here. Then again I might be imagining a specific scope that is different from how you understand it.

Reply:
Consider this nominally resolved: I'd leave it to you to decide if the refactoring suggested was too much to fit here.

Reply:
> I think we can limit the scope of this refactoring to only mouse_click_* function that uses this same pattern. In other words, it will affect the existing `mouse_click_checkbox` and no more.

This is the only function with the exact same pattern, however there are many (basically all) mouse_click_* functions that have a very similar pattern.  They seemed similar enough to me that a refactor like this could be applied to many more of the mouseclick functions.  In that sense I think I was imagining a larger scope for this refactor.  
I spent a bit of time trying to work on this, and still have my scratch code saved for later, but decided to put it on pause and deal with later as the scope was kind of expanding as I was working on it (e.g. as I was adding it i realized if I just changed it slightly then such and such other mouse_click functions could also utilize it, etc.)